Revise Versioning Structure v3.4.0➔v3.4 and current➔next

nate-double-u opened the pull request:
Moving files.

Updating: config.toml content/docs/next/_index.md content/docs/next/metrics.md content/docs/v3.4/_index.md layouts/index.redirects layouts/partials/docs/hero.html layouts/partials/docs/nav-panel.html layouts/partials/home/hero.html

Closes: https://github.com/etcd-io/website/issues/152

/cc @chalin

nate-double-u commented:
Deploy previews:
* https://deploy-preview-161--etcd.netlify.app/
* https://deploy-preview-161--etcd.netlify.app/current
* https://deploy-preview-161--etcd.netlify.app/latest
* https://deploy-preview-161--etcd.netlify.app/
* https://deploy-preview-161--etcd.netlify.app/docs/v3.4.0/
* https://deploy-preview-161--etcd.netlify.app/docs/next

nate-double-u commented:
I wonder if we should move what was `current` into `v3.4` now. Current/next has the updated descriptions that were introduced when we migrated the docs over from etcd-io/etcd, and I think current/next represents the latest `v3.4` version.

@ptabor @spzala @jberkus @chalin

nate-double-u commented:
- [ ] I need to update the deprecation warning.

chalin commented on layouts/index.redirects:
```diff
@@ -1,6 +1,9 @@
 /docs/latest/*     /docs/{{ $latest }}/:splat
+/docs/current     /docs/next/
+/docs/current/*     /docs/next/:splat
+/docs/v2    /docs/v2.3/
 /docs/v2    /docs/v2.3/
```
Dup'd line

nate-double-u replied:
thanks, good catch

nate-double-u commented on layouts/partials/home/hero.html:
```diff
@@ -17,7 +17,7 @@
 
       <div class="buttons is-centered">
-        <a class="button is-primary is-inverted is-medium is-radiusless" href="/docs/current/">
+        <a class="button is-primary is-inverted is-medium is-radiusless" href="/docs/latest/">
```
I'm not happy that this breaks the button on `localhost`, even when using `netlify dev` this doesn't forward correctly (it brings you to http://localhost:8888/docs/), but it does work on the deploy preview/will work on the production build.

chalin replied:
Hmm, it works for me.

chalin replied:
Oh, maybe not ... wait.

chalin replied:
I can confirm that it works for me locally using `netlify dev`

nate-double-u replied:
I verify your confirmation. Thanks!

nate-double-u replied:
No, I take back my verification -- the frontpage button on `localhost:8888` takes me to http://localhost:8888/docs/, not http://localhost:8888/docs/v3.4/

nate-double-u commented:
Corrected nav:
* https://deploy-preview-161--etcd.netlify.app/docs/v2.3/libraries-and-tools/
  Note, next no longer links to v2.3 and correctly links to the top level of next.

Corrected deprecation warning:
* https://deploy-preview-161--etcd.netlify.app/docs/next
